cedar-policy/src/api/err.rs

```diff
@@ -449,12 +448,12 @@ pub struct ValidationError {
 impl ValidationError {
     /// Extract details about the exact issue detected by the validator.
     pub fn error_kind(&self) -> &ValidationErrorKind {
-        self.error.error_kind()
+        self.error.kind()
     }
 
-    /// Extract the location where the validator found the issue.
-    pub fn location(&self) -> &SourceLocation {
-        SourceLocation::ref_cast(self.error.location())
+    /// Extract the policy id of the policy where the validator found the issue.
+    pub fn policy_id(&self) -> &PolicyId {
+        PolicyId::ref_cast(self.error.policy_id())
     }
```

Comment:
Callers previously had the ability to get `range_start()` and `range_end()`.  Are we intentionally removing that here (and from `ValidationWarning`)?

Reply:
It's still `Diagnostic`, so it just requires a bit of extra work. I don't want to keep those functions here but not the rest of our errors.

Reply:
Good point